From d546a68744d2f98caac8db7ea08eb4c71f23a7cd Mon Sep 17 00:00:00 2001
From: Simon McVittie <simon.mcvittie@collabora.co.uk>
Date: Tue, 25 Oct 2016 19:42:07 +0100
Subject: [PATCH 5/6] qemu: put eofcat in an executable tmpfs on
 /run/autopkgtest

This avoids having to write it to /bin, which might be read-only.

Signed-off-by: Simon McVittie <simon.mcvittie@collabora.co.uk>
---
 virt/autopkgtest-virt-qemu | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/virt/autopkgtest-virt-qemu b/virt/autopkgtest-virt-qemu
index 24e4353..98f0317 100755
--- a/virt/autopkgtest-virt-qemu
+++ b/virt/autopkgtest-virt-qemu
@@ -218,7 +218,9 @@ def setup_shared(shared_dir):
 
     term = VirtSubproc.get_unix_socket(os.path.join(workdir, 'ttyS1'))
 
-    term.send(b'''mkdir -p -m 1777 /run/autopkgtest/shared
+    term.send(b'''mkdir -p /run/autopkgtest
+mount -t tmpfs run-autopkgtest /run/autopkgtest
+mkdir -p -m 1777 /run/autopkgtest/shared
 mount -t 9p -o trans=virtio,access=any autopkgtest /run/autopkgtest/shared
 chmod 1777 /run/autopkgtest/shared
 touch /run/autopkgtest/shared/done_shared
@@ -240,7 +242,7 @@ touch /run/autopkgtest/shared/done_shared
     # arg), or an "exit flag" file exists.
     # We don't run that from /run/autopkgtest/shared as 9p from older QEMU
     # versions is buggy and causes "invalid numeric result" errors on that.
-    term.send(b'''PYTHON=$(which python3) || PYTHON=$(which python); cat <<EOF > /bin/eofcat; chmod 755 /bin/eofcat
+    term.send(b'''PYTHON=$(which python3) || PYTHON=$(which python); cat <<EOF > /run/autopkgtest/eofcat; chmod 755 /run/autopkgtest/eofcat
 #!$PYTHON
 import sys, os, fcntl, time, errno
 (feof, fexit) = sys.argv[1:]
@@ -376,7 +378,7 @@ t_stderr.start()
 # "real" stdin behaviour.
 s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
 s.connect('%(tty)s')
-cmd = 'PYTHONHASHSEED=0 /bin/eofcat %%(d)s/stdin_eof %%(d)s/exit.tmp < %%(d)s/stdin | ' \\
+cmd = 'PYTHONHASHSEED=0 /run/autopkgtest/eofcat %%(d)s/stdin_eof %%(d)s/exit.tmp < %%(d)s/stdin | ' \\
       '(%%(c)s >> %%(d)s/stdout 2>> %%(d)s/stderr; echo $? > %%(d)s/exit.tmp);' \\
       'mv %%(d)s/exit.tmp %%(d)s/exit\\n' %% \\
        {'d': job_guest, 'c': ' '.join(map(pipes.quote, sys.argv[1:]))}
-- 
2.10.1

